d2/src/main/java/com/linkedin/d2/balancer/clusterfailout/ClusterFailoutBucketConfig.java

```diff
@@ -0,0 +1,131 @@
+package com.linkedin.d2.balancer.clusterfailout;
+
+import java.util.Map;
+import java.util.Date;
+
+/**
+ * This class is a simple data structure for tracking the traffic state of a given application.
+ */
+
+
+/*
+failedoutBuckets: array[BucketFailoutConfig] {
+    fabric: FabricUrn
+    partition: string
+    // e.g. buckets 0-4 are offline
+    bucketIdRange: IntRange
+    offlineAt: Time
+    onlineAt: optional Time
+  }
+ */
+public class ClusterFailoutBucketConfig
+{
+  // Fabric that this configuration is addressing.
+  private final String _fabricUrn;
+  // Traffic partition, eg. Main, Guest, Static, etc.
+  private final String _partition;
+  // Range of buckets offline. Eg, 5, so buckets 0-4 are offline.
+  private final Integer _bucketOfflineRange;
+  // What time should the buckets go offline at?
+  private final Date _offlineAt;
+  // What time should they come back online?
+  private final Date _onlineAt;
+
+  private final static String FABRIC_URN_PROPERTY = "fabric";
+  private final static String PARTITION_PROPERTY = "partition";
+  private final static String BUCKET_OFFLINE_RANGE_PROPERTY = "bucketOfflineRange";
+  private final static String OFFLINE_AT_PROPERTY = "offlineAt";
+  private final static String ONLINE_AT_PROPERTY = "onlineAt";
+
+  public ClusterFailoutBucketConfig(String fabricUrn,
+      String partition,
+      Integer bucketOfflineRange,
+      Date offlineAt,
+      Date onlineAt)
+  {
+    _fabricUrn = fabricUrn;
+    _partition = partition;
+    _bucketOfflineRange = bucketOfflineRange;
+    _offlineAt = offlineAt;
+    _onlineAt = onlineAt;
+  }
+
+  public static ClusterFailoutBucketConfig createFromMap(Map<String, Object> configMap) {
+    ClusterFailoutBucketConfig clusterFailoutBucketConfig = null;
+    try {
+      clusterFailoutBucketConfig = new ClusterFailoutBucketConfig((String) configMap.get(ClusterFailoutBucketConfig.FABRIC_URN_PROPERTY),
```

Comment:
you can just do `return new Cluster...` so that you don't need to have a placeholder variable outside the try block